jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.AsUndirectedGraph;
+
+import java.util.*;
+
+/**
+ * Implementation of the lexicographical breadth-first search algorithm for chordal graph recognition.
+ * <p>
+ * A chordal graph is one in which all cycles of four or more vertices have a chord, which is an edge that
+ * is not part of the cycle but connects two vertices of the cycle.
+ * <p>
+ * For more information on the topic see the following
+ * <a href="http://www.cse.iitd.ac.in/~naveen/courses/CSL851/uwaterloo.pdf">article</a>:
+ * <i>"CS 762: Graph-theoretic algorithms.
+ * Lecture notes of a graduate course. University of Waterloo. Fall 1999, Winter 2002, Winter 2004."</i>
+ * <p>
```

Comment:
Since that doc covers a huge number of topics, including one of the more specific references from your previous comments would be good.